doc/kuadrantctl-ci-cd.md

```diff
@@ -0,0 +1,184 @@
+# kuadrantctl - CI/CD with Tekton and Argo CD
+
+This guide demonstrates setting up a CI/CD pipeline using Tekton to deploy Kubernetes Gateway API and Kuadrant resources generated by `kuadrantctl`, from an OpenAPI specification. In this example, these resources are applied directly to the cluster where Tekton is running.
+
+Prerequisites:
+
+- (Tekton Pipelines[https://tekton.dev/]) installed on your Kubernetes or OpenShift cluster.
+- (`kubectl`[https://kubernetes.io/docs/reference/kubectl/]) configured to communicate with your cluster (i.e you have a kubectl config available with access to your cluster)
+- (Tekton CLI `tkn`[https://tekton.dev/docs/cli/]) (optional) for easier interaction with Tekton resources.
+
+Setup:
+
+First, create a dedicated namespace:
+
+```bash
+kubectl create namespace petstore
+```
+
+Step 1: Create a Persistent Volume Claim
+
+To store Tekton build artifacts, create a PVC in the petstore namespace:
+
+```bash
+kubectl apply -n petstore -f - <<EOF
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: tekton-kuadrantctl-pvc
+  namespace: petstore
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+EOF
+```
+
+Step 2: Define the Tekton Task
+
+Define the task that outlines steps to clone a repository, generate Kuadrant and Kubernetes resources using `kuadrantctl`, and apply them directly to the cluster:
+
+```bash
+kubectl apply -f - <<'EOF'
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: run-kuadrantctl
+  namespace: petstore
+spec:
+  params:
+    - name: gitRepoUrl
+      description: URL of the git repository to clone
+    - name: gitRevision
+      description: Git revision to checkout (branch, tag, sha)
+  workspaces:
+    - name: source
+      description: Workspace to checkout the git repo
+    - name: kubeconfig
+      description: Workspace containing kubeconfig for Kubernetes cluster access
+  steps:
+    - name: clean-workspace
+      image: alpine:latest
+      script: |
+        sh -c 'rm -rf $(workspaces.source.path)/* $(workspaces.source.path)/.[!.]* $(workspaces.source.path)/..?*'
+    - name: clone
+      image: alpine/git:latest
+      script: |
+        git clone $(params.gitRepoUrl) $(workspaces.source.path)
+        cd $(workspaces.source.path)
+        git checkout $(params.gitRevision)
+    - name: download-kuadrantctl
+      image: curlimages/curl:latest
+      script: |
+        ARCH=$(uname -m)
+        case $ARCH in
+        x86_64) BIN_ARCH="amd64";;
+        arm64) BIN_ARCH="arm64";;
+        aarch64) BIN_ARCH="arm64";;
+        *) echo "Unsupported architecture: $ARCH" && exit 1 ;;
+        esac
+        cd $(workspaces.source.path)
+        curl -LO "https://github.com/Kuadrant/kuadrantctl/releases/download/v0.2.0/kuadrantctl-v0.2.0-linux-$BIN_ARCH.tar.gz"
+        tar -xzf kuadrantctl-v0.2.0-linux-$BIN_ARCH.tar.gz
+    - name: run-kuadrantctl
+      image: alpine:latest
```

Comment:
might be worth considering a kudrantctl image at some point with this stuff baked in.Could merge/shorten steps 3/4